tor/src/main/scala/org/bitcoins/tor/client/TorClient.scala

```diff
@@ -0,0 +1,231 @@
+package org.bitcoins.tor.client
+
+import grizzled.slf4j.Logging
+import org.bitcoins.core.util.EnvUtil
+import org.bitcoins.tor.TorProtocolHandler._
+import org.bitcoins.tor.config.TorAppConfig
+import org.bitcoins.tor.{Socks5ProxyParams, TorParams}
+
+import java.io.{File, FileNotFoundException}
+import java.net.InetSocketAddress
+import java.nio.file.{Files, Path, StandardCopyOption}
+import scala.concurrent.{ExecutionContext, Future}
+import scala.sys.process.{Process, ProcessBuilder}
+import scala.util.{Failure, Success, Try}
+
+/** A trait that helps start bitcoind/eclair when it is started via bitcoin-s */
+class TorClient(implicit ec: ExecutionContext, conf: TorAppConfig)
+    extends Logging {
+
+  private[this] var processOpt: Option[Process] = None
+
+  private lazy val process: ProcessBuilder = scala.sys.process.Process(cmd)
+
+  lazy val socks5ProxyParams: Socks5ProxyParams = conf.socks5ProxyParams match {
+    case Some(params) => params
+    case None =>
+      val addr = InetSocketAddress.createUnresolved(
+        "127.0.0.1",
+        Socks5ProxyParams.DefaultPort)
+      Socks5ProxyParams(
+        address = addr,
+        credentialsOpt = None,
+        randomizeCredentials = true
+      )
+  }
+
+  lazy val torParams: TorParams = conf.torParams match {
+    case Some(params) => params
+    case None =>
+      val control = InetSocketAddress.createUnresolved(
+        "127.0.0.1",
+        TorParams.DefaultControlPort)
+
+      val auth = SafeCookie()
+      val privKeyPath = conf.datadir.resolve("tor_priv_key")
+
+      TorParams(control, auth, privKeyPath)
+  }
+
+  private lazy val authenticationArg = torParams.authentication match {
+    case Password(_) =>
+      "error"
+//      s"--HashedControlPassword $password" // todo: need to hash the password correctly
```

Comment:
is this still TODO? I'm not sure if I understand what is happening here. Perhaps a log message?

Reply:
Yeah, I can't seem to find how to correctly hash the password, made it throw an error now